src/config.rs

```diff
@@ -52,10 +52,25 @@ macro_rules! create_config {
             $(pub $i: $ty),+
         }
 
+        #[derive(RustcDecodable, Clone)]
+        pub struct ParsedConfig {
```

Comment:
Could you add a comment please explaining what this struct is for and how the parsing the process works